NEWS.md

```diff
@@ -1,18 +1,102 @@
-3.31.2
+3.32.1
 ======
 
+## Highlights
+
+ * `mongodb()`: add `workers()` support (multi-threaded connection pooling)
+   
+   The MongoDB driver now supports the `workers()` option, which specifies the
+   number of parallel workers to be used.
+   Workers are based on the connection pooling feature of the MongoDB C library.
+   
+   This increases the throughput of the MongoDB destination driver.
+   
+   Example:
+   
+   ```
+   destination {
+     mongodb(
+       uri("mongodb://hostA,hostB/syslog?replicaSet=my_rs&wtimeoutMS=10000&socketTimeoutMS=10000&connectTimeoutMS=10000&serverSelectionTimeoutMS=5000")
+       collection("messages")
+       workers(8)
+     );
+   };
+   ```
+   ([#3621](https://github.com/syslog-ng/syslog-ng/pull/3621))
+ * `mongodb()`: template support for the `collection()` option
+   
+   The `collection()` option of the MongoDB destination driver now accepts
+   templates, for example:
+   
+   ```
+   destination {
+     mongodb(
+       uri("mongodb://host/syslog?wtimeoutMS=10000&socketTimeoutMS=10000&connectTimeoutMS=10000&serverSelectionTimeoutMS=5000")
+       collection("${HOST}_messages")
+       workers(8)
+     );
+   };
+   ```
+   ([#3621](https://github.com/syslog-ng/syslog-ng/pull/3621))
+
+## Features
+
+ * `time-reopen`: Support the `time-reopen()` option on the driver level for the following drivers:
+    * sources: `example-diskq-source`, `python-fetcher`
+    * destinations: `amqp`, `example-destination`, `file`, `http`, `mongodb`, `network`, `pipe`,
+                    `program`, `pseudofile`, `python`, `redis`, `riemann`, `smtp`, `sql`, `stomp`,
+                    `syslog`, `tcp`, `tcp6`, `udp`, `udp6`, `unix-dgram`, `unix-stream`, `usertty`
+   ([#3585](https://github.com/syslog-ng/syslog-ng/pull/3585))
+ * `csv-parser()`: add drop-invalid() option along with the already existing
+   flag with the same name. This is to improve the consistency of the
+   configuration language.
+   ([#3547](https://github.com/syslog-ng/syslog-ng/pull/3547))
+ * `usertty() destination`: Support changing the terminal disable timeout with the `time-reopen()` option.
+   Default timeout change to 60 from 600. If you wish to use the old 600 timeout, add `time-reopen(600)`
+   to your config in the `usertty()` driver.
+   ([#3585](https://github.com/syslog-ng/syslog-ng/pull/3585))
+ * `syslog-parser()`: add a new drop-invalid() option that allows the use of
+   syslog-parser() in if statements. Normally a syslog-parser() injects an
```

Comment:
This sounds exciting, don't we want to add it to the Highlights section?

(The `collection()` MongoDB option is not such a big deal, we can replace that if you want to.)

Reply:
If this is a vote I would keep the template support in mongodb's `collection` option in the Highlights section. :)

Reply:
The `syslog-parser()` new flag is a nice feature, but I do not think a lot of people tinker on that level with the configuration. The template support likely more common for people to have, and it also matches with the other highlight marking this release as a mongodb improvements kinda release. (Sure there are other nice feature it has as well.)